index.js

```diff
@@ -528,80 +574,140 @@ Client.prototype._destroy = function (err, cb) {
     clearTimeout(this._corkTimer)
     this._corkTimer = null
   }
-  clients[this._index] = null // remove global reference to ease garbage collection
+  clientsToAutoEnd[this._index] = null // remove global reference to ease garbage collection
   this._chopper.destroy()
   this._agent.destroy()
   cb(err)
 }
 
-function onStream (client, onerror) {
```

Comment:
note to reviewer: A total re-write of `onStream` -- now called `getChoppedStreamHandler` and `makeIntakeRequest` --  is below (after the _getBackoffDelay() function). There is large block comment that should mostly explain it.